MiddleweightReflection/IMrHasCustomAttributes.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System.Collections.Immutable;
+
+namespace MiddleweightReflection
+{
+    public interface IMrHasCustomAttributes : IMrNamedElement
```

Comment:
Why the inheritance? Having custom attributes doesn't imply having a name. How about just an MrBase that has the 3 methods and is derived from by all 5 types?